### PR TITLE
spec for view types

### PR DESCRIPTION
**Pull request by Araq**
Spec for https://github.com/nim-lang/RFCs/issues/178

### CODE REVIEW

File: doc/manual_experimental.rst [hunk 1]
```diff
@@ -1814,18 +1814,49 @@ For example:
...
+**Note**:  ``--experimental:views`` is more effective
+with ``--experimental:strictFuncs``.
+
+A view type is a type that is or contains one of the following types:
 
 - ``var T`` (mutable view into ``T``)
 - ``lent T`` (immutable view into ``T``)
 - ``openArray[T]`` (pair of (pointer to array of ``T``, size))
 
+For exampe:
```
Comment (ghost) [inline doc/manual_experimental.rst]:
```suggestion
For example:
```

typo :)

File: doc/manual_experimental.rst [hunk 2]
```diff
@@ -1814,18 +1814,49 @@ For example:
...
+.. code-block:: nim
+
+  type
+    NotView1 = proc (x: openArray[int])
+    NotView2 = ptr openArray[char]
+    NotView3 = ptr array[4, var int]
+
+
+A *mutable* view type is a type that is or contains a ``var T`` type,
+an *immutable* view type is a view type that is not a mutable view type.
```
Comment (ghost) [inline doc/manual_experimental.rst]: maybe rephrase this?

Comment (Araq) [reply doc/manual_experimental.rst]: Why?

Comment (KeepCoolWithCoolidge) [reply doc/manual_experimental.rst]: I think Yardanico is trying to point out that something is wrong but can't quite put his finger on it. Here you have two independent clauses joined together with a comma, which is technically incorrect. You can fix this by inserting a "while" or "but" after the comma, replacing the comma with an "and," replacing the comma with a semicolon, or replacing the comma with a period/full stop and making the second clause a new sentence. I suggest the first option because it emphasizes contrast.
